### ISSUE TITLE
Android 12: Cannot setup storage access framework

### ISSUE THREAD

**Issue by demetry14** [d4rken-org/sdmaid-se#639]
Cannot access Android/obb when attempting to setup storage access framework on Android 12.  I read that was an issue on Android 13, but I am on a Galaxy S10 and they max out at Android 12.

Click setup > Grant Access > Android > "all you see is media"

[Screenshot_20230919-010126_SD Maid SE](https://github.com/d4rken-org/sdmaid-se/assets/5199569/8a9d13d8-4235-463b-abca-6e702a9248c0)

[Screenshot_20230919-010113_Files](https://github.com/d4rken-org/sdmaid-se/assets/5199569/f8593ccf-23ce-4298-9e5c-2382f8a5517e)

**Comment by demetry14** [comment]
Also note the folder exists as I have Solid Explorer and can access them.
![Screenshot_20230919-010859_Solid Explorer](https://github.com/d4rken-org/sdmaid-se/assets/5199569/a3adbcdc-e01c-45ff-97e2-cbcbf4b155b5)

![Screenshot_20230919-010905_Solid Explorer](https://github.com/d4rken-org/sdmaid-se/assets/5199569/6ce70418-df68-4ebb-9056-d29f6a443f81)

**Comment by KryptoKing93** [comment]
Hooe this gets resolved soon. I'm facing the same issue

**Comment by demetry14** [comment]
Hopefully this helps as well.

[eu.darken.sdmse_logfile_1695108257091.log.zip](https://github.com/d4rken-org/sdmaid-se/files/12675163/eu.darken.sdmse_logfile_1695108257091.log.zip)

**Comment by Pawwwle** [comment]
I also have this error on two Android 12 devices.

**Comment by d4rken** [comment]
Is the issue accessing `Android/data` or `Android/obb` or both?

Currently it looks like Samsung's latest OneUI update backported some Android 13 changes to Android 12.

> Solid Explorer and can access them.

Hm will look into that, if Solid Explorer has found a workaround we could do that too. It's possible they are faking access though, or "simulating" it. You can guess some of the content and display it as "access". Can Solid Explorer write to `data` or `obb`?

**Comment by Pawwwle** [comment]
Boths. The problems occurs on my Motorola smartphone and Lenovo tablet.

**Comment by demetry14** [comment]
> Is the issue accessing `Android/data` or `Android/obb` or both?
> 
> Currently it looks like Samsung's latest OneUI update backported some Android 13 changes to Android 12.
> 
> > Solid Explorer and can access them.
> 
> Hm will look into that, if Solid Explorer has found a workaround we could do that too. It's possible they are faking access though, or "simulating" it. You can guess some of the content and display it as "access". Can Solid Explorer write to `data` or `obb`?

I have no issues accessing Android/data, only Android/obb.

Android/data was accessible right away and I verified by clearing app data and rechecking. 

![Screenshot_20230921-202627_SD Maid SE](https://github.com/d4rken-org/sdmaid-se/assets/5199569/a07bcbf4-bb0a-458a-8dd1-d7faffdb3a59)

**Comment by 268Vladimir** [comment]
Android 12 (A11+), NO root(!), MIUI 14, Redme Note 9S, global, stock.
No access to Android/data (obb).
Wiche turns SDMaid SE (SDMaid) into a regular cache cleaning program . ?!

**Comment by 268Vladimir** [comment]
![IMG_20230925_082951](https://github.com/d4rken-org/sdmaid-se/assets/145930779/20878c61-f3ce-4abd-b75c-cc2eeade3645)

**Comment by 268Vladimir** [comment]
I followed the recommendations, the result was zero.  Documents from system applications in A11+ are excluded.
![IMG_20230925_162106](https://github.com/d4rken-org/sdmaid-se/assets/145930779/caa6e4b5-96ee-4cdf-a61c-76a6d17b5fbc)
![IMG_20230925_153210](https://github.com/d4rken-org/sdmaid-se/assets/145930779/6fd9fb2b-a0d8-4450-a1df-f7797c1321c3)

**Comment by JohannesOe** [comment]
Samsung OneUi 3.1 Android 11 same problem. Only "media" folder shows up

**Comment by 268Vladimir** [comment]
Solved a problem 
https://4pda.to/forum/index.php?showtopic=1066732&view=findpost&p=125295743

**Comment by MeloriTensei** [comment]
@demetry14 @Pawwwle @KryptoKing93 Hi, please rest assured that we (me and possibly the community with Darken) are currently looking into how other apps got around this.  This issue has been present since Samsung's July 2023 Security patch because they brought the restrictions from Android 13 to Android 12.

**Comment by d4rken** [comment]
TL;DR: Google Play updated the `Files` app. The newer `Files` app has backported some restrictions from Android 13 and 14. 

Workaround: Uninstall the update to the `Files` app and then set up SD Maid.

See #703 for a detailed explanation.
